File: backend/src/main/java/com/erkang/domain/entity/MedicalRecord.java
package com.erkang.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 病历实体
 * _Requirements: 5.1, 5.3_
 */
@Data
@TableName("medical_record")
public class MedicalRecord {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long consultationId;
    private Long patientId;
    private Long doctorId;
    private String recordNo;
    
    private String chiefComplaint;      // 主诉
    private String presentIllness;      // 现病史
    private String pastHistory;         // 既往史
    private String allergyHistory;      // 过敏史
    private String physicalExam;        // 体格检查
    private String auxiliaryExam;       // 辅助检查
    private String diagnosis;           // 初步诊断
    private String treatmentPlan;       // 处理建议
    private String followupAdvice;      // 随访建议
    
    private String aiSummary;           // AI生成摘要
    private Integer aiConfirmed;        // AI摘要是否已确认
    
    private String status;              // DRAFT/SUBMITTED
    private LocalDateTime submittedAt;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/services/medicalRecord.ts
/**
 * 病历服务
 * _Requirements: 6.7_
 */

import { get, post, put } from './http';
import type { Id } from '@/types';

// 病历类型
export interface MedicalRecord {
  id: Id;
  consultationId: Id;
  patientId: Id;
  doctorId: Id;
  chiefComplaint?: string;
  presentIllness?: string;
  pastHistory?: string;
  allergies?: string;
  physicalExam?: string;
  diagnosis?: string;
  treatment?: string;
  advice?: string;
  createdAt: string;
  updatedAt?: string;
}

// 创建/更新病历请求
export interface SaveMedicalRecordRequest {
  consultationId: Id;
  chiefComplaint?: string;
  presentIllness?: string;
  pastHistory?: string;
  allergies?: string;
  physicalExam?: string;
  diagnosis?: string;
  treatment?: string;
  advice?: string;
}

/**
 * 病历服务
 * _Requirements: 6.7_
 */
export const medicalRecordService = {
  /**
   * 获取病历详情
   */
  getByConsultation: async (consultationId: Id): Promise<MedicalRecord | null> => {
    try {
      const response = await get<MedicalRecord>(`/medical-records/consultation/${consultationId}`);
      return response.data;
    } catch {
      return null;
    }
  },

  /**
   * 保存病历（创建或更新）
   * _Requirements: 6.7_
   */
  save: async (data: SaveMedicalRecordRequest): Promise<MedicalRecord> => {
    const response = await post<MedicalRecord>('/medical-records', data);
    return response.data;
  },

  /**
   * 更新病历
   */
  update: async (id: Id, data: Partial<SaveMedicalRecordRequest>): Promise<MedicalRecord> => {
    const response = await put<MedicalRecord>(`/medical-records/${id}`, data);
    return response.data;
  },
};

export default medicalRecordService;
